### PR TITLE
[runtime] Merge two identical code paths

### PR DESCRIPTION
**Pull request by spouliot**

### CODE REVIEW

File: runtime/trampolines.m [hunk 1]
```diff
@@ -80,8 +80,6 @@
 	/* Any changes in this method probably need to be reflected in the static registrar as well */
 	switch (type [0]) {
 		case _C_CLASS:
```
Comment (wachaudh) [inline runtime/trampolines.m]: shouldn't the case be removed all together?

Comment (wachaudh) [inline runtime/trampolines.m]: shouldn't something be returned in this

File: runtime/trampolines.m [hunk 2]
```diff
@@ -80,8 +80,6 @@
...
-			return xamarin_get_handle_for_inativeobject (retval, exception_gchandle);
```
Comment (wachaudh) [inline runtime/trampolines.m]: Isn't the function expected to return some sort of void pointer? If the function hits this empty case it will never return.

Comment (spouliot) [reply runtime/trampolines.m]: it's not an empty case, i.e. there's no `break` or `continue`, so it just share the same code path as `_C_SEL`

Comment (mandel-macaque) [reply runtime/trampolines.m]: @wachaudh that is what they mean when they say a switch case will fall-through.

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

**Pull request merged**